### ISSUE TITLE
AssertionError in `lambda.EventSourceMapping` for version 6.13

### ISSUE THREAD

**Issue by chrispyduck** [pulumi/pulumi-aws#3092]
### What happened?

I'm testing an upgrade of the python pulumi-aws from 5.10 to 6.13 and i'm seeing an `AssertionError: Unexpected type. Expected 'list' got '<class 'str'>'` that I believe I've narrowed down to uses of `aws.lambda_.EventSourceMapping` resources (by moving a return statement around to find the resource that causes it to fail), e.g.:

```python
aws.lambda_.EventSourceMapping(
    "queue",
    event_source_arn=queue.arn,
    function_name=lambda_.arn,
    opts=opts,
)
```

I've verified that `queue.arn` and `lambda_.arn` are strings and that the documentation and SDK code both list these parameters as `str` not `list`, and this code was working as-is before the upgrade, so I suspect this is a bug. 

### Example

I'm trying to create a minimal project to reproduce this issue but I haven't been able to, yet. I'll update once I figure it out. For now all I can report is that every occurrence of `EventSourceMapping` has this problem in my (private) repo.

### Output of `pulumi about`

```
CLI          
Version      3.94.0
Go Version   go1.21.3
Go Compiler  gc

Plugins
NAME        VERSION
aws         6.13.0
aws-native  0.90.0
datadog     4.23.0
python      unknown

Host     
OS       darwin
Version  14.1.1
Arch     arm64

This project is written in python: executable='./env/bin/python' version='3.9.16'

Backend        
Name           pulumi.com
URL            *removed*
User           *removed*
Organizations  *removed*
Token type     personal

Dependencies:
NAME                     VERSION
blinker                  1.7.0
boltons                  20.2.1
boto3                    1.28.17
datadog-api-client       2.19.0
dnspython                2.4.2
Flask                    2.2.3
hiredis                  2.2.3
inflection               0.5.1
launchdarkly-server-sdk  7.6.1
lxml                     4.9.3
orjson                   3.9.10
pip                      22.3.1
pulumi-aws               6.13.0
pulumi-aws-native        0.90.0
pulumi-datadog           4.23.0
pydantic                 2.5.2
redis                    4.6.0
regret                   0.45.2
retry                    0.9.2
rich                     13.7.0
sentry-sdk               1.38.0
setuptools               69.0.2
tomli                    2.0.1
tomlkit                  0.12.3
```

### Additional context

I was hoping to work around this using `aws-native`'s version of this resource but I ran into this instead: https://github.com/pulumi/pulumi-aws-native/issues/1038. This is preventing me from upgrading 

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by chrispyduck** [comment]
Per @t0yv0 this may be related to #3075

**Comment by VenelinMartinov** [comment]
Thank you for reporting @chrispyduck, looks like the issue in the native provider has a workaround - https://github.com/pulumi/pulumi-aws-native/issues/1038#issuecomment-1681238307 - did you try that? It might be the quickest way to unblocking your use case.

**Comment by t0yv0** [comment]
Getting another report of this stack trace from a customer. Not yet certain which resource.

**Comment by VenelinMartinov** [comment]
Got a repro:
```python
import pulumi
import pulumi_aws as aws
import pulumi_archive as archive

assume_role = aws.iam.get_policy_document(
    statements=[
        aws.iam.GetPolicyDocumentStatementArgs(
            effect="Allow",
            principals=[
                aws.iam.GetPolicyDocumentStatementPrincipalArgs(
                    type="Service",
                    identifiers=["lambda.amazonaws.com"],
                )
            ],
            actions=["sts:AssumeRole"],
        )
    ]
)
iam_for_lambda = aws.iam.Role(
    "iamForLambda",
    assume_role_policy=assume_role.json,
    managed_policy_arns=["arn:aws:iam::aws:policy/AmazonSQSFullAccess"],
)
lambda_ = archive.get_file(
    type="zip", source_file="lambda.js", output_path="lambda_function_payload.zip"
)
test_lambda = aws.lambda_.Function(
    "testLambda",
    code=pulumi.FileArchive("lambda_function_payload.zip"),
    role=iam_for_lambda.arn,
    handler="index.test",
    runtime="nodejs18.x",
)

queue = aws.sqs.Queue("queue")


esm = aws.lambda_.EventSourceMapping(
    "example", event_source_arn=queue.arn, function_name=test_lambda.arn
)

```

steps:
1. `pulumi up` with `pulumi-aws` version `5.42.0`.
2. `pulumi up` with `pulumi-aws` version `6.13.1`.
3. Observe failure.

**Comment by VenelinMartinov** [comment]
Looks like the schema for the `queue` parameter for `EventSourceMapping` changed:  
In `5.43.0` it was:
```
"queues": {
                    "type": "array",
                    "items": {
                        "type": "string"
                    },
                    "description": "The name of the Amazon MQ broker destination queue to consume. Only available for MQ sources. A single queue name must be specified.\n"
                },
```

And now it is:
```
"queues": {
                    "type": "string",
                    "description": "The name of the Amazon MQ broker destination queue to consume. Only available for MQ sources. The list must contain exactly one queue name.\n"
                },
```

I guess it was always meant to have only one parameter but the change of the type is a breaking change. Specifically in the repro above, when we call `"/pulumirpc.ResourceMonitor/RegisterResource"` for `EventSourceMapping`, we used to pass `"queues": []` but now we pass `"queues":null`. Passing `[]` triggers the exception.

**Comment by t0yv0** [comment]
I'm wondering if the intent of this change https://github.com/pulumi/pulumi-terraform-bridge/commit/f0a3ffcaf34a05be5484f7b0aaf15e2b77f212b7 is making it worse here, in a sense that the bridge pretends that `[]->null` translation is done implicitly but does not inform the engine..

**Comment by VenelinMartinov** [comment]
We are still working on a fix but the issue has a workaround - one can specify a value for `queues`. For example:
```python
esm = aws.lambda_.EventSourceMapping(
    "example",
    event_source_arn=queue.arn,
    function_name=test_lambda.arn,
    queues="queue",
    opts=pulumi.ResourceOptions(delete_before_replace=True),
)
```
This will force the resource to update and work around the issue.
NOTE: This will cause a REPLACEMENT of the resource, which could trigger replacements of any resources which depend on it.
Also note that `delete_before_replace` is needed here.

**Comment by VenelinMartinov** [comment]
This was released in `v6.13.3`
